librubyfmt/src/format.rs

```diff
@@ -708,20 +727,35 @@ pub fn format_method_call(ps: &mut dyn ConcreteParserState, method_call: MethodC
             };
 
             if !args.is_empty() {
-                ps.breakable_of(
-                    delims,
-                    Box::new(|ps| {
+                match get_def_expression_from_args(&args) {
+                    Some(_) => {
+                        // If we match `def ...` as the first argument, just
+                        // emit it without any delimiters.
                         ps.with_formatting_context(
                             FormattingContext::ArgsList,
                             Box::new(|ps| {
-                                format_list_like_thing(ps, args, false);
-                                ps.emit_collapsing_newline();
+                                ps.emit_space();
+                                format_list_like_thing(ps, args, true);
```

Comment:
if instead of using format_list_like_thing_here, if you just directly format the def, it should fix the excessive newline problem.